src/main/java/io/molnarsandor/trelloclone/project/controller/ProjectController.java

```diff
@@ -0,0 +1,69 @@
+package io.molnarsandor.trelloclone.project.controller;
+
+import io.molnarsandor.trelloclone.global_exceptions.CustomGlobalExceptionResponse;
+import io.molnarsandor.trelloclone.global_exceptions.ValidationErrorExceptionResponse;
+import io.molnarsandor.trelloclone.project.exceptions.ProjectIdExceptionResponse;
+import io.molnarsandor.trelloclone.project.exceptions.ProjectNotFoundExceptionResponse;
+import io.molnarsandor.trelloclone.project.model.ProjectDTO;
+import io.molnarsandor.trelloclone.user.exceptions.UserNotLoggedInExceptionResponse;
+import io.molnarsandor.trelloclone.util.DeleteDTO;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.BindingResult;
+import springfox.documentation.annotations.ApiIgnore;
+
+import java.security.Principal;
+import java.util.List;
+
+public interface ProjectController {
```

Comment:
Nagyon jó megoldás szerintem, ha szétszeded a controllereket interfész és osztály implementációkra. Elkülönülnek a dokumentációk és a valós tennivalók. 

Amit még esetleg hozzátennék, az az, hogy az interfészekben is tudod használni a Springes `@RequestMapping` és társai annotációt, ha még tovább szeretnéd az implementációt "szépíteni" vagy "tisztábbá" tenni, akkor a mappingeket is ide raknám át, illetve a `@Valid` és `@RequestBody` annotációkat is kirakhatod a megfelelő paraméterekre, utána pedig az implementációban már csak a tiszta szolgáltatás hívások maradnak.